sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/SpannerChangestreamsReadSchemaTransformProvider.java

```diff
@@ -193,15 +218,19 @@ public abstract static class Builder {
     }
   }
 
-  private static final class DataChangeRecordToRow extends DoFn<DataChangeRecord, Row> {
+  public static final class DataChangeRecordToRow extends DoFn<DataChangeRecord, Row> {
```

Comment:
why is this changed to public?

Reply:
In order to test it in the SpannerChangestreamsReadDlqTest. There didn't seem a straight forward way to test the dlq. So I tested the DoFn itself.

Reply:
got it. Can you annotate it as @VisibleForTesting then?

Reply:
Also, you should create a test case with the scenario that artificially forces use of the DLQ, it shouldn't remain untested

Reply:
My explanation was not clear, the dlq is tested in a way, just not the end to end workflow. I provide the DoFn with a correct and incorrect record. Then check if they are tagged properly.

Reply:
ty

Reply:
Done (I have added @VisibleForTesting).